tests/amazon/aws/sensors/test_s3_sensors.py

```diff
@@ -243,6 +245,32 @@ def test_s3_key_sensor_with_wildcard_async(self, mock_hook, mock_poke, context):
 
         assert isinstance(exc.value.trigger, S3KeyTrigger), "Trigger is not a S3KeyTrigger"
 
+    def test_soft_fail(self):
+        """Raise AirflowSkipException in case soft_fail is true"""
+        sensor = S3KeySensorAsync(
+            task_id="s3_key_sensor_async", bucket_key="key", bucket_name="bucket", soft_fail=True
+        )
+        with pytest.raises(AirflowSkipException):
+            sensor.execute_complete(
+                context={}, event={"status": "error", "message": "mocked error", "soft_fail": True}
+            )
+
+    @pytest.mark.parametrize(
+        "soft_fail,exec",
```

Comment:
```suggestion
        "soft_fail,exception",
```
looks like a typo here: `exec`